code/datums/ai/kitchenbot/kitchenbot_controller.dm

```diff
@@ -0,0 +1,123 @@
+/datum/ai_controller/kitchenbot
+	ai_movement = /datum/ai_movement/basic_avoidance
+	movement_delay = 0.3 SECONDS
+	blackboard = list(
+	BB_KITCHENBOT_MODE = KITCHENBOT_MODE_REFUSE,
+	//BB_KITCHENBOT_MODE = KITCHENBOT_MODE_OFF,
+	BB_KITCHENBOT_CHOSEN_DISPOSALS = null,
+	BB_KITCHENBOT_FAILED_LAST_TARGET_SEARCH = FALSE,
+	BB_KITCHENBOT_TARGET_TO_DISPOSE = null,
+	BB_KITCHENBOT_CHOSEN_GRIDDLE = null,
+	BB_KITCHENBOT_CHOSEN_STOCKPILE = null,
+	BB_KITCHENBOT_ITEMS_WATCHED = list(),
+	BB_KITCHENBOT_ITEMS_BANNED = list(),
+	BB_KITCHENBOT_TAKE_OFF_GRILL = list(),
+	BB_KITCHENBOT_TARGET_IN_STOCKPILE = null
+	)
+
+
+/datum/ai_controller/kitchenbot/TryPossessPawn(atom/new_pawn)
+	if(!istype(new_pawn, /mob/living/simple_animal/bot/kitchenbot))
+		return AI_CONTROLLER_INCOMPATIBLE
+	return ..() //Run parent at end
+
+/datum/ai_controller/kitchenbot/SelectBehaviors(delta_time)
+	current_behaviors = list()
+	switch(blackboard[BB_KITCHENBOT_MODE])
```

Comment:
This would ideally not be a state machine but with the scope of the AI and performance considerations it's fine, I might convert it to be more clever about switching later but only if I think its good for the overall gameplay.